doctor_visits/delphi_doctor_visits/process_data.py

```diff
@@ -0,0 +1,162 @@
+"""Module providing functions for processing and wrangling data."""
+
+from datetime import datetime
+from pathlib import Path
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+
+from .config import Config
+
+
+def format_outname(prefix: str, se: bool, weekday: bool):
+    """
+    Write out results.
+
+    Parameters
+    ----------
+    prefix:
+    se: boolean to write out standard errors, if true, use an obfuscated name
+    weekday: boolean for weekday adjustments.
+             signals will be generated with weekday adjustments (True) or without
+                adjustments (False)
+
+    Returns
+    -------
+    outname str
+    """
+    out_name = "smoothed_adj_cli" if weekday else "smoothed_cli"
+    if se:
+        assert prefix is not None, "template has no obfuscated prefix"
+        out_name = prefix + "_" + out_name
+    return out_name
+
+
+def format_df(df: pd.DataFrame, geo_id: str, se: bool, logger):
+    """
+    Format dataframe and checks for anomalies to write results.
+
+    Parameters
+    ----------
+    df: dataframe from output from update_sensor
+    geo_id: geographic resolution, one of ["county", "state", "msa", "hrr", "nation", "hhs"]
+    se: boolean to write out standard errors, if true, use an obfuscated name
+    logger
+
+    Returns
+    -------
+    filtered and formatted dataframe
+    """
+    # report in percentage
+    df["val"] = df["val"] * 100
+    df["se"] = df["se"] * 100
+
+    val_isnull = df["val"].isnull()
+    df_val_null = df[val_isnull]
+    if not df_val_null.empty:
+        logger.info("sensor value is nan, check pipeline")
```

Comment:
question: so previously these were `assert` statements that would halt the pipeline. The pro is that kept potential bad values out of our db, the con is the pipeline could get tanked by one bad row. The new behavior is reversed. Do we want to keep the old behavior or have the new behavior? I vaguely prefer the old way, given how tough it is to patch things for us in general, though maybe that changes with #1977. thoughts?

Reply:
1977 makes patching faster, but it's still the same process at its core. I agree we should keep the assert.